main/src/ca/uwaterloo/flix/language/phase/Kinder.scala

```diff
@@ -812,6 +812,10 @@ object Kinder {
       mapN(exp1Val, exp2Val) {
         case (exp1, exp2) => KindedAst.Expression.FixpointProject(pred, exp1, exp2, Type.freshVar(Kind.Star, loc.asSynthetic), loc)
       }
+
+    case ResolvedAst.Expression.Error(m) =>
+      KindedAst.Expression.Error(m).toSuccess // TODO: Refactor to SoftFailure
```

Comment:
I think we need a version of toSoftFailure that works with the empty list (since the error is already in the Validation).

Reply:
Can we add that and use it everywhere?

Reply:
Yes but maybe that should be in a different PR because right now each phase has to be refactored to use `CompilationMessage` as the error type of the `Validation` instead of `KindError` or `WeederError`

Reply:
Why?

Reply:
If the List is empty it should be OK?

Reply:
Oh that's true of course but I figured we wanted `m` to be in the errors list

Reply:
No, because its already there (!)

But indeed its a bit strange.